bazel: `pkg/sql:sql_test` fails w/ a circular dependency error when run under the race detector

There's no error when run without the race detector.

```
root@c7f4bbcfdab1:/cockroach# bazel test --config ci --config race //pkg/sql:sql_test --test_env=COCKROACH_LOGIC_TESTS_SKIP=true --test_env=GOMAXPROCS=8
INFO: Analyzed target //pkg/sql:sql_test (1076 packages loaded, 25933 targets configured).
INFO: Found 1 test target...
WARNING: failed to create one or more convenience symlinks for prefix '_bazel/':
  cannot create symbolic link _bazel/bin -> /root/.cache/bazel/_bazel_root/cc377fc379544923cc7508dd261e4a48/execroot/cockroach/bazel-out/k8-fastbuild/bin:  /cockroach/_bazel/bin (Read-only file system)
  cannot create symbolic link _bazel/testlogs -> /root/.cache/bazel/_bazel_root/cc377fc379544923cc7508dd261e4a48/execroot/cockroach/bazel-out/k8-fastbuild/testlogs:  /cockroach/_bazel/testlogs (Read-only file system)
  cannot create symbolic link bazel-out -> /root/.cache/bazel/_bazel_root/cc377fc379544923cc7508dd261e4a48/execroot/cockroach/bazel-out:  /cockroach/bazel-out (Read-only file system)
  cannot create symbolic link _bazel/out -> /root/.cache/bazel/_bazel_root/cc377fc379544923cc7508dd261e4a48/execroot/cockroach/bazel-out:  /cockroach/_bazel/out (Read-only file system)
  cannot create symbolic link _bazel/cockroach -> /root/.cache/bazel/_bazel_root/cc377fc379544923cc7508dd261e4a48/execroot/cockroach:  /cockroach/_bazel/cockroach (Read-only file system)
ERROR: /root/.cache/bazel/_bazel_root/cc377fc379544923cc7508dd261e4a48/external/io_bazel_rules_go/BUILD.bazel:44:7: GoStdlib external/io_bazel_rules_go/stdlib_/pkg failed: (Exit 1): builder failed: error executing command bazel-out/host/bin/external/go_sdk/builder stdlib -sdk external/go_sdk -installsuffix linux_amd64_race -tags ignore,bazel,crdb_test,gss,bazel,crdb_test,gss,race -out ... (remaining 2 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox builder failed: error executing command bazel-out/host/bin/external/go_sdk/builder stdlib -sdk external/go_sdk -installsuffix linux_amd64_race -tags ignore,bazel,crdb_test,gss,bazel,crdb_test,gss,race -out ... (remaining 2 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports net/http
	imports crypto/tls
	imports crypto/x509
	imports archive/tar: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports bytes: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/crypto/tls/key_agreement.go:9:2: found packages md5 (example_test.go) and main (gen.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/crypto/md5
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/net/http/client.go:14:2: found packages tls (alert.go) and main (generate_cert.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/crypto/tls
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/crypto/tls/cipher_suites.go:16:2: found packages x509 (cert_pool.go) and main (root_ios_gen.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/crypto/x509
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports errors: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports strconv
	imports flag: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports fmt: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/ast: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/format: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports go/scanner: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports go/token: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports net/http
	imports crypto/tls
	imports crypto/md5
	imports text/template
	imports internal/fmtsort: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports net/http
	imports context
	imports internal/reflectlite: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports strconv
	imports log
	imports os
	imports io/fs: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports strconv
	imports log
	imports os
	imports internal/poll
	imports internal/syscall/unix
	imports syscall
	imports log: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports strconv
	imports math: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/math/fma.go:7:8: found packages bits (bits.go) and main (make_examples.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/math/bits
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/math/big/int.go:12:2: found packages rand (example_test.go) and main (gen_cooked.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/math/rand
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/time/genzabbrs.go:22:2: found packages http (alpn_test.go) and main (triv.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/net/http
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports strconv
	imports log
	imports os
	imports internal/poll
	imports internal/syscall/unix
	imports syscall
	imports os: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/net/http/triv.go:18:2: found packages exec (bench_test.go) and main (read3.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/os/exec
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports reflect: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/internal/reflectlite/value.go:9:2: found packages runtime (alg.go) and main (mkduff.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/runtime
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/runtime/internal/math/math.go:7:8: found packages sys (arch.go) and main (gengoos.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/runtime/internal/sys
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/go/token/position.go:9:2: found packages sort (example_interface_test.go) and main (genzfunc.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/sort
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/encoding/xml/marshal.go:14:2: found packages strconv (atob.go) and main (makeisprint.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/strconv
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/internal/syscall/unix/at.go:10:2: found packages syscall (creds_test.go) and main (mkasm.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/syscall
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/io/fs/fs.go:12:2: found packages time (example_test.go) and main (genzabbrs.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/time
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports bytes: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/crypto/tls/key_agreement.go:9:2: found packages md5 (example_test.go) and main (gen.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/crypto/md5
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/net/http/client.go:14:2: found packages tls (alert.go) and main (generate_cert.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/crypto/tls
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/crypto/tls/cipher_suites.go:16:2: found packages x509 (cert_pool.go) and main (root_ios_gen.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/crypto/x509
found packages gob (codec_test.go) and main (decgen.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/encoding/gob
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports errors: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports strconv
	imports flag: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports fmt: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/ast: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/go/build/build.go:12:2: found packages doc (comment.go) and main (headscan.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/go/doc
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/go/build/build.go:12:2: found packages doc (comment.go) and main (headscan.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/go/doc
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/format: import cycle not allowed
package go/importer
	imports go/internal/gccgoimporter
	imports go/types
	imports go/importer: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/go/internal/gccgoimporter/gccgoinstallation.go:9:2: found packages types (api.go) and main (gotype.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/go/types
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports go/scanner: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports go/token: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/go/internal/gccgoimporter/gccgoinstallation.go:9:2: found packages types (api.go) and main (gotype.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/go/types
found packages main (gen.go) and palette (generate.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/image/color/palette
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/image/draw/draw.go:14:2: import "image/internal/imageutil" is a program, not an importable package
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/image/gif/writer.go:14:2: import "image/color/palette" is a program, not an importable package
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/image/draw/draw.go:14:2: found packages main (gen.go) and imageutil (imageutil.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/image/internal/imageutil
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/image/jpeg/reader.go:13:2: import "image/internal/imageutil" is a program, not an importable package
found packages suffixarray (example_test.go) and main (gen.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/index/suffixarray
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports net/http
	imports crypto/tls
	imports crypto/md5
	imports text/template
	imports internal/fmtsort: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports net/http
	imports context
	imports internal/reflectlite: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports strconv
	imports log
	imports os
	imports io/fs: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports strconv
	imports log
	imports os
	imports internal/poll
	imports internal/syscall/unix
	imports syscall
	imports log: import cycle not allowed
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports strconv
	imports math: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/math/fma.go:7:8: found packages bits (bits.go) and main (make_examples.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/math/bits
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/math/big/int.go:12:2: found packages rand (example_test.go) and main (gen_cooked.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/math/rand
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/time/genzabbrs.go:22:2: found packages http (alpn_test.go) and main (triv.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/net/http
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports strconv
	imports log
	imports os
	imports internal/poll
	imports internal/syscall/unix
	imports syscall
	imports os: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/net/http/triv.go:18:2: found packages exec (bench_test.go) and main (read3.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/os/exec
package archive/tar
	imports bytes
	imports errors
	imports internal/reflectlite
	imports runtime
	imports flag
	imports fmt
	imports internal/fmtsort
	imports reflect
	imports math
	imports math/bits
	imports go/format
	imports go/ast
	imports go/scanner
	imports go/token
	imports sort
	imports go/parser
	imports io/fs
	imports time
	imports encoding/xml
	imports reflect: import cycle not allowed
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/internal/reflectlite/value.go:9:2: found packages runtime (alg.go) and main (mkduff.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/runtime
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/runtime/internal/math/math.go:7:8: found packages sys (arch.go) and main (gengoos.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/runtime/internal/sys
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/go/token/position.go:9:2: found packages sort (example_interface_test.go) and main (genzfunc.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/sort
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/encoding/xml/marshal.go:14:2: found packages strconv (atob.go) and main (makeisprint.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/strconv
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/internal/syscall/unix/at.go:10:2: found packages syscall (creds_test.go) and main (mkasm.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/syscall
bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/io/fs/fs.go:12:2: found packages time (example_test.go) and main (genzabbrs.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/time
found packages main (generate_zipdata.go) and tzdata (tzdata.go) in bazel-out/k8-opt-exec-2B5CBBC6-ST-1516a2ab7dba/bin/external/io_bazel_rules_go/stdlib_/src/time/tzdata
```

Epic CRDB-8036

--- a/pkg/sql/schemachanger/scop/BUILD.bazel
+++ b/pkg/sql/schemachanger/scop/BUILD.bazel
@@ -24,7 +24,7 @@ go_library(
 go_binary(
     name = "gen-visitors",
     srcs = ["generate_visitor.go"],
-    gotags = ["ignore"],
+    gotags = ["generator"],
     deps = [
         "//pkg/cli/exit",
         "@com_github_cockroachdb_errors//:errors",

--- a/pkg/sql/schemachanger/scop/generate_visitor.go
+++ b/pkg/sql/schemachanger/scop/generate_visitor.go
@@ -8,7 +8,7 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-// +build ignore
+// +build generator
 
 package main
 

